crates/starknet-devnet-server/src/api/json_rpc/endpoints.rs

```diff
@@ -376,14 +376,19 @@ impl JsonRpcHandler {
             .parse::<usize>()
             .map_err(|_| ApiError::InvalidContinuationToken)?;
 
-        let (events, has_more_events) = starknet.get_events(
-            filter.from_block,
-            filter.to_block,
-            filter.address,
-            filter.keys,
-            page * filter.chunk_size,
-            Some(filter.chunk_size),
-        )?;
+        let (events, has_more_events) = starknet
+            .get_events(
+                filter.from_block,
+                filter.to_block,
+                filter.address,
+                filter.keys,
+                page * filter.chunk_size,
+                Some(filter.chunk_size),
+            )
+            .map_err(|err| match err {
+                Error::NoBlock => ApiError::BlockNotFound,
```

Comment:
Why not do this error mapping in the same place as other json-rpc method implementations?

Reply:
From what I can see, internal errors map to API errors in this layer (endpoints.rs). Could you please clarify your question?

Reply:
You're right, I don't know what I was thinking. Perhaps I was just confused by the wildcard match, but now that I've inspected the error types, it really seems like another way to do it. Resolving this.